git/status.go

```diff
@@ -346,6 +346,11 @@ func StatusLong(c *Client, files []File, untracked StatusUntrackedMode, linepref
 				if err != nil {
 					return "", err
 				}
+				// TODO instead of filtering afterwards, perhaps the untracked should be filtered at the source
+				matches, _ := CheckIgnore(c, CheckIgnoreOptions{}, []File{fname})
```

Comment:
Why don't we put this check in LsFiles and include ExcludeStandard in the opts? That would go a lot farther than anything a filter in every porcelain command